### ISSUE TITLE
Specifying output option -o overrides --format

### ISSUE THREAD

**Issue by CoveredInChocolate** [ManimCommunity/manim#2321]
## Description of bug / unexpected behavior
I am running on Ubuntu, using:
* Manim Community v0.12.0
* Python 3.8.5

When I include the options `--format=gif -o fileName.gif`, it creates an mp4 file.

## Expected behavior
I expect it to create a gif.


## How to reproduce the issue
Just a simple code I think I got from the Quickstart documentation. The issue is unrelated to the code as far as I can tell.

```py
from manim import *

class SquareToCircle(Scene):
    def construct(self):
        circle = Circle()                    # create a circle
        circle.set_fill(PINK, opacity=0.5)   # set color and transparency

        square = Square()                    # create a square
        square.flip(RIGHT)                   # flip horizontally
        square.rotate(-3 * TAU / 8)          # rotate a certain amount

        self.play(Create(square))            # animate the creation of the square
        self.play(Transform(square, circle)) # interpolate the square into the circle
        self.play(FadeOut(square))           # fade out animation
```

Console output
```
(base) vdaxx@ubserv:~/MANIM/project$ manim scene.py --format=gif -o outputBug.gif -ql SquareToCircle
Manim Community v0.12.0

[11/24/21 21:49:33] INFO     Animation 0 : Using cached data (hash :                               cairo_renderer.py:74
                             3163782288_2875658055_1793580042)                                                         
                    INFO     Animation 1 : Using cached data (hash :                               cairo_renderer.py:74
                             2201830969_208498043_1793580042)                                                          
                    INFO     Animation 2 : Using cached data (hash :                               cairo_renderer.py:74
                             2201830969_3021192394_1793580042)                                                         
                    INFO     Combining to Movie file.                                          scene_file_writer.py:604
                    INFO                                                                       scene_file_writer.py:718
                             File ready at '/home/vdaxx/MANIM/project/media/videos/scene/480p                         
                             15/outputBug.mp4'                                                                         
                                                                                                                       
                    INFO     Rendered SquareToCircle                                                       scene.py:234
                             Played 3 animations  
```



## Additional media files
N/A


## Logs
<details><summary>Terminal output</summary>
<!-- Add "-v DEBUG" when calling manim to generate more detailed logs -->

```
(base) vdaxx@ubserv:~/MANIM/project$ manim scene.py --format=gif -o outputBug.gif -ql SquareToCircle -v DEBUG
Manim Community v0.12.0

[11/24/21 21:54:56] DEBUG    Hashing ...                                                                 hashing.py:344
                    DEBUG    Hashing done in 0.009578 s.                                                 hashing.py:356
                    DEBUG    Hash generated :  3163782288_2875658055_1793580042                          hashing.py:359
                    INFO     Animation 0 : Using cached data (hash :                               cairo_renderer.py:74
                             3163782288_2875658055_1793580042)                                                         
                    DEBUG    List of the first few animation hashes of the scene:                  cairo_renderer.py:83
                             ['3163782288_2875658055_1793580042']                                                      
                    DEBUG    Hashing ...                                                                 hashing.py:344
                    DEBUG    Hashing done in 0.010422 s.                                                 hashing.py:356
                    DEBUG    Hash generated :  2201830969_208498043_1793580042                           hashing.py:359
                    INFO     Animation 1 : Using cached data (hash :                               cairo_renderer.py:74
                             2201830969_208498043_1793580042)                                                          
                    DEBUG    List of the first few animation hashes of the scene:                  cairo_renderer.py:83
                             ['3163782288_2875658055_1793580042',                                                      
                             '2201830969_208498043_1793580042']                                                        
                    DEBUG    Hashing ...                                                                 hashing.py:344
                    DEBUG    Hashing done in 0.005314 s.                                                 hashing.py:356
                    DEBUG    Hash generated :  2201830969_3021192394_1793580042                          hashing.py:359
                    INFO     Animation 2 : Using cached data (hash :                               cairo_renderer.py:74
                             2201830969_3021192394_1793580042)                                                         
                    DEBUG    List of the first few animation hashes of the scene:                  cairo_renderer.py:83
                             ['3163782288_2875658055_1793580042',                                                      
                             '2201830969_208498043_1793580042',                                                        
                             '2201830969_3021192394_1793580042']                                                       
                    INFO     Combining to Movie file.                                          scene_file_writer.py:604
                    DEBUG    Partial movie files to combine (3 files): ['/home/vdaxx/MANIM/pr scene_file_writer.py:547
                             oject/media/videos/scene/480p15/partial_movie_files/SquareToCircl                         
                             e/3163782288_2875658055_1793580042.mp4', '/home/vdaxx/MANIM/proj                         
                             ect/media/videos/scene/480p15/partial_movie_files/SquareToCircle/                         
                             2201830969_208498043_1793580042.mp4', '/home/vdaxx/MANIM/project                         
                             /media/videos/scene/480p15/partial_movie_files/SquareToCircle/220                         
                             1830969_3021192394_1793580042.mp4']                                                       
                    INFO                                                                       scene_file_writer.py:718
                             File ready at '/home/vdaxx/MANIM/project/media/videos/scene/480p                         
                             15/outputBug.mp4'                                                                         
                                                                                                                       
                    INFO     Rendered SquareToCircle                                                       scene.py:234
                             Played 3 animations  
```


## System specifications

<details><summary>System Details</summary>

- OS (with version, e.g Windows 10 v2004 or macOS 10.15 (Catalina)):
```
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.2 LTS
Release:	20.04
Codename:	focal
```
- RAM: 16GB
- Python version: Python 3.8.5
- Installed modules (provide output from `pip list`):
```
Package                            Version
---------------------------------- -------------------
alabaster                          0.7.12
anaconda-client                    1.7.2
anaconda-navigator                 1.10.0
anaconda-project                   0.8.3
argh                               0.26.2
argon2-cffi                        20.1.0
asn1crypto                         1.4.0
astroid                            2.4.2
astropy                            4.0.2
async-generator                    1.10
atomicwrites                       1.4.0
attrs                              20.3.0
autopep8                           1.5.4
Babel                              2.8.1
backcall                           0.2.0
backports.functools-lru-cache      1.6.1
backports.shutil-get-terminal-size 1.0.0
backports.tempfile                 1.0
backports.weakref                  1.0.post1
beautifulsoup4                     4.9.3
bitarray                           1.6.1
bkcharts                           0.2
bleach                             3.2.1
bokeh                              2.2.3
boto                               2.49.0
Bottleneck                         1.3.2
brotlipy                           0.7.0
certifi                            2020.6.20
cffi                               1.14.3
chardet                            3.0.4
click                              7.1.2
click-default-group                1.2.2
cloudpickle                        1.6.0
cloup                              0.7.1
clyent                             1.2.2
colorama                           0.4.4
colour                             0.1.5
commonmark                         0.9.1
conda                              4.9.2
conda-build                        3.20.5
conda-package-handling             1.7.2
conda-verify                       3.4.2
contextlib2                        0.6.0.post1
cryptography                       3.1.1
cycler                             0.10.0
Cython                             0.29.21
cytoolz                            0.11.0
dask                               2.30.0
decorator                          5.1.0
defusedxml                         0.6.0
diff-match-patch                   20200713
distributed                        2.30.1
docutils                           0.16
entrypoints                        0.3
et-xmlfile                         1.0.1
fastcache                          1.1.0
filelock                           3.0.12
flake8                             3.8.4
Flask                              1.1.2
fsspec                             0.8.3
future                             0.18.2
gevent                             20.9.0
glcontext                          2.3.4
glob2                              0.7
gmpy2                              2.0.8
greenlet                           0.4.17
h5py                               2.10.0
HeapDict                           1.0.1
html5lib                           1.1
idna                               2.10
imageio                            2.9.0
imagesize                          1.2.0
importlib-metadata                 2.0.0
iniconfig                          1.1.1
intervaltree                       3.1.0
ipykernel                          5.3.4
ipython                            7.21.0
ipython-genutils                   0.2.0
ipywidgets                         7.5.1
isort                              5.6.4
isosurfaces                        0.1.0
itsdangerous                       1.1.0
jdcal                              1.4.1
jedi                               0.18.0
jeepney                            0.5.0
Jinja2                             2.11.2
joblib                             0.17.0
json5                              0.9.5
jsonschema                         3.2.0
jupyter                            1.0.0
jupyter-client                     6.1.7
jupyter-console                    6.2.0
jupyter-core                       4.6.3
jupyter-manim                      1.3
jupyterlab                         2.2.6
jupyterlab-pygments                0.1.2
jupyterlab-server                  1.2.0
keyring                            21.4.0
kiwisolver                         1.3.0
lazy-object-proxy                  1.4.3
libarchive-c                       2.9
llvmlite                           0.34.0
locket                             0.2.0
lxml                               4.6.1
manim                              0.12.0
manimlib                           0.2.0
ManimPango                         0.3.1
mapbox-earcut                      0.12.11
MarkupSafe                         1.1.1
matplotlib                         3.3.2
mccabe                             0.6.1
mistune                            0.8.4
mkl-fft                            1.2.0
mkl-random                         1.1.1
mkl-service                        2.3.0
mock                               4.0.2
moderngl                           5.6.4
moderngl-window                    2.4.0
more-itertools                     8.6.0
mpmath                             1.1.0
msgpack                            1.0.0
multipledispatch                   0.6.0
navigator-updater                  0.2.1
nbclient                           0.5.1
nbconvert                          6.0.7
nbformat                           5.0.8
nest-asyncio                       1.4.2
networkx                           2.5
nltk                               3.5
nose                               1.3.7
notebook                           6.1.4
numba                              0.51.2
numexpr                            2.7.1
numpy                              1.20.1
numpydoc                           1.1.0
olefile                            0.46
opencv-python                      4.5.1.48
openpyxl                           3.0.5
packaging                          20.4
pandas                             1.1.3
pandocfilters                      1.4.3
parso                              0.8.1
partd                              1.1.0
path                               15.0.0
pathlib2                           2.3.5
pathtools                          0.1.2
patsy                              0.5.1
pep8                               1.7.1
pexpect                            4.8.0
pickleshare                        0.7.5
Pillow                             8.0.1
pip                                20.2.4
pkginfo                            1.6.1
plotly                             4.14.3
pluggy                             0.13.1
ply                                3.11
progressbar                        2.5
prometheus-client                  0.8.0
prompt-toolkit                     3.0.16
psutil                             5.7.2
ptyprocess                         0.6.0
py                                 1.9.0
pycairo                            1.20.0
pycodestyle                        2.6.0
pycosat                            0.6.3
pycparser                          2.20
pycurl                             7.43.0.6
pydocstyle                         5.1.1
pydub                              0.25.0
pyflakes                           2.2.0
pyglet                             1.5.21
Pygments                           2.7.2
pylint                             2.6.0
pyodbc                             4.0.0-unsupported
pyOpenSSL                          19.1.0
pyparsing                          2.4.7
pyrr                               0.10.3
pyrsistent                         0.17.3
PySocks                            1.7.1
pytest                             0.0.0
python-dateutil                    2.8.1
python-jsonrpc-server              0.4.0
python-language-server             0.35.1
pytz                               2020.1
PyWavelets                         1.1.1
pyxdg                              0.27
PyYAML                             5.3.1
pyzmq                              19.0.2
QDarkStyle                         2.8.1
QtAwesome                          1.0.1
qtconsole                          4.7.7
QtPy                               1.9.0
regex                              2020.10.15
requests                           2.24.0
retrying                           1.3.3
rich                               6.2.0
rope                               0.18.0
Rtree                              0.9.4
ruamel-yaml                        0.15.87
scikit-image                       0.17.2
scikit-learn                       0.23.2
scipy                              1.6.1
screeninfo                         0.6.7
seaborn                            0.11.0
SecretStorage                      3.1.2
Send2Trash                         1.5.0
setuptools                         50.3.1.post20201107
simplegeneric                      0.8.1
singledispatch                     3.4.0.3
sip                                4.19.13
six                                1.15.0
skia-pathops                       0.7.1
snowballstemmer                    2.0.0
sortedcollections                  1.2.1
sortedcontainers                   2.2.2
soupsieve                          2.0.1
Sphinx                             3.2.1
sphinxcontrib-applehelp            1.0.2
sphinxcontrib-devhelp              1.0.2
sphinxcontrib-htmlhelp             1.0.3
sphinxcontrib-jsmath               1.0.1
sphinxcontrib-qthelp               1.0.3
sphinxcontrib-serializinghtml      1.1.4
sphinxcontrib-websupport           1.2.4
spyder                             4.1.5
spyder-kernels                     1.9.4
SQLAlchemy                         1.3.20
statsmodels                        0.12.0
sympy                              1.6.2
tables                             3.6.1
tblib                              1.7.0
terminado                          0.9.1
testpath                           0.4.4
threadpoolctl                      2.1.0
tifffile                           2020.10.1
toml                               0.10.1
toolz                              0.11.1
tornado                            6.0.4
tqdm                               4.59.0
traitlets                          5.0.5
typing-extensions                  3.7.4.3
ujson                              4.0.1
unicodecsv                         0.14.1
urllib3                            1.25.11
watchdog                           0.10.3
wcwidth                            0.2.5
webencodings                       0.5.1
Werkzeug                           1.0.1
wheel                              0.35.1
widgetsnbextension                 3.5.1
wrapt                              1.11.2
wurlitzer                          2.0.1
xlrd                               1.2.0
XlsxWriter                         1.3.7
xlwt                               1.3.0
xmltodict                          0.12.0
yapf                               0.30.0
zict                               2.0.0
zipp                               3.4.0
zope.event                         4.5.0
zope.interface                     5.1.2
```

## Additional comments
Don't think ffmpeg or LaTeX have anything to do with this.

**Comment by CoveredInChocolate** [comment]
Oh, I should also mention that I am quite new to manim, so please forgive me if I somehow messed up!

**Comment by Kiran-Raj-Dev** [comment]
Fixed via #2528. Closing
